runtime/stdlib/account.go

```diff
@@ -791,6 +800,236 @@ func newPublicAccountContractsValue(
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			typeParameterPair := invocation.TypeParameterTypes.Oldest()
+			if typeParameterPair == nil {
+				panic(errors.NewUnreachableError())
+			}
+
+			ty := sema.NewCapabilityType(gauge, typeParameterPair.Value)
+			publishedType := publishedValue.Value.StaticType(invocation.Interpreter)
+			if !inter.IsSubTypeOfSemaType(publishedType, ty) {
+				panic(interpreter.ForceCastTypeMismatchError{
+					ExpectedType:  ty,
+					ActualType:    inter.MustConvertStaticToSemaType(publishedType),
+					LocationRange: getLocationRange(),
+				})
+			}
+
+			handler.EmitEvent(
+				inter,
+				AccountInboxRemoveEventType,
+				[]interpreter.Value{
+					providerValue,
+					providerValue,
+					nameValue,
+				},
+				getLocationRange,
+			)
+
+			value := publishedValue.Value.Transfer(
+				inter,
+				getLocationRange,
+				atree.Address(address),
+				true,
+				nil,
+			)
+
+			inter.WriteStored(address, inboxStorageDomain, nameValue.Str, nil)
+
+			return interpreter.NewSomeValueNonCopying(inter, value)
+		},
+		sema.AuthAccountTypeInboxPublishFunctionType,
+	)
+}
+
+func accountInboxClaimFunction(
+	gauge common.MemoryGauge,
+	handler EventEmitter,
+	address common.Address,
+	recipientValue interpreter.AddressValue,
+) *interpreter.HostFunctionValue {
+	return interpreter.NewHostFunctionValue(
+		gauge,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			nameValue, ok := invocation.Arguments[0].(*interpreter.StringValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			providerValue, ok := invocation.Arguments[1].(interpreter.AddressValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			inter := invocation.Interpreter
+			getLocationRange := invocation.GetLocationRange
+
+			providerAddress := providerValue.ToAddress()
+
+			readValue := inter.ReadStored(providerAddress, inboxStorageDomain, nameValue.Str)
+			if readValue == nil {
+				return interpreter.NewNilValue(gauge)
+			}
+			publishedValue := readValue.(*interpreter.PublishedValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			// compare the intended recipient with the caller
+			if !publishedValue.Recipient.Equal(inter, getLocationRange, recipientValue) {
+				return interpreter.NewNilValue(gauge)
+			}
+
+			typeParameterPair := invocation.TypeParameterTypes.Oldest()
+			if typeParameterPair == nil {
+				panic(errors.NewUnreachableError())
+			}
+
+			ty := sema.NewCapabilityType(gauge, typeParameterPair.Value)
+			publishedType := publishedValue.Value.StaticType(invocation.Interpreter)
+			if !inter.IsSubTypeOfSemaType(publishedType, ty) {
+				panic(interpreter.ForceCastTypeMismatchError{
+					ExpectedType:  ty,
+					ActualType:    inter.MustConvertStaticToSemaType(publishedType),
+					LocationRange: getLocationRange(),
+				})
+			}
+
+			value := publishedValue.Value.Transfer(
+				inter,
+				getLocationRange,
+				atree.Address(address),
```

Comment:
Wrong address, value is transferred from storage to stack:

```suggestion
				atree.Address{},
```